### PR TITLE
Add a gallery example showing the usage of text symbols

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

This PR adds a gallery example that shows the usage of text symbols via ``Figure.plot()``.

Text symbols gallery example

**Preview** at https://pygmt-git-gallery-text-symbols-gmt.vercel.app/gallery/symbols/text_symbols.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
Is "text symbols" different from normal texts? Why not using the `Figure.text()`?

**Comment by michaelgrund** [comment]
> 
> 
> Is "text symbols" different from normal texts? Why not using the `Figure.text()`?

I expanded the docstring and added a further sentence that is used in the GMT docs. In my opinion we should at least mention that it's possible to plot text symbols also via ``Figure.plot()``.

File: examples/gallery/symbols/text_symbols.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+r"""
+Text symbols
+------------
+The :meth:`pygmt.Figure.plot` method allows to plot text symbols. Text is
+normally placed with the :meth:`pygmt.Figure.text` method but there are times
+we wish to treat a character of even a string as plottable symbol.
```
Comment (seisman) [inline examples/gallery/symbols/text_symbols.py:6]:
```suggestion
we wish to treat a character or even a string as a plottable symbol.
```

File: examples/gallery/symbols/text_symbols.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+A text symbol can be drawn by passing **l**\ *size*\ **+t**\ *string* to
+the ``style`` parameter where *size* defines the size of the text symbol
+(note: the size is only approximate; no individual scaling is done for
+different characters) and *string* can be a letter or a text string
+(less than 256 characters). Optionally, you can append **+f**\ *font* to
+select a particular font [Default is FONT_ANNOT_PRIMARY] as well as
```
Comment (seisman) [inline examples/gallery/symbols/text_symbols.py:12]:
```suggestion
select a particular font [Default is :gmt-term:`FONT_ANNOT_PRIMARY`] as well as
```

File: examples/gallery/symbols/text_symbols.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+**+j**\ *justify* to change the justification [Default is CM]. Outline
+and fill color of the text symbols can be customized via the ``pen``
+and ``color`` parameters, respectively.
+
+For all supported octal codes and fonts see the GMT cookbook
+https://docs.generic-mapping-tools.org/latest/cookbook/octal-codes.html and
+https://docs.generic-mapping-tools.org/latest/cookbook/postscript-fonts.html.
```
Comment (seisman) [inline examples/gallery/symbols/text_symbols.py:19]:
```suggestion
:gmt-docs:`cookbook/octal-codes.html` and
:gmt-docs:`cookbook/postscript-fonts.html`.
```

**Review by seisman — APPROVED** [review]
The example looks good to me except a few suggestions.

**Review by weiji14 — APPROVED** [review]
Had an idea to spell out 'πgmt', but ok with this example too :laughing:

**Comment by michaelgrund** [comment]
> 
> 
> Had an idea to spell out 'πgmt', but ok with this example too 😆

Nice idea but I think it's not possible to mix Symbols and standard text here.

**Pull request merged**